src/WrappedMToken.sol

```diff
@@ -164,55 +179,35 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
 
     /// @inheritdoc IWrappedMToken
     function startEarningFor(address account_) external {
-        _revertIfNotApprovedEarner(account_);
-
         if (!isEarningEnabled()) revert EarningIsDisabled();
 
-        Account storage accountInfo_ = _accounts[account_];
-
-        if (accountInfo_.isEarning) return;
-
         // NOTE: Use `currentIndex()` if/when upgrading to support `startEarningFor` while earning is disabled.
-        uint128 currentIndex_ = _currentMIndex();
-
-        accountInfo_.isEarning = true;
-        accountInfo_.lastIndex = currentIndex_;
+        _startEarningFor(account_, _currentMIndex());
+    }
 
-        uint240 balance_ = accountInfo_.balance;
+    /// @inheritdoc IWrappedMToken
+    function startEarningFor(address[] calldata accounts_) external {
+        if (!isEarningEnabled()) revert EarningIsDisabled();
 
-        _addTotalEarningSupply(balance_, currentIndex_);
+        uint128 currentIndex_ = _currentMIndex();
 
-        unchecked {
-            totalNonEarningSupply -= balance_;
+        for (uint256 index_; index_ < accounts_.length; ++index_) {
+            _startEarningFor(accounts_[index_], currentIndex_);
         }
-
-        emit StartedEarning(account_);
     }
 
     /// @inheritdoc IWrappedMToken
     function stopEarningFor(address account_) external {
-        _revertIfApprovedEarner(account_);
+        _stopEarningFor(account_, currentIndex());
+    }
 
+    /// @inheritdoc IWrappedMToken
+    function stopEarningFor(address[] calldata accounts_) external {
```

Comment:
seems optional, can live in periphery multicall contracts or EarnerManager if its a primary goal for adding this looping logic here.